src/toxcore/tcp/client/client.rs

```diff
@@ -1179,18 +1191,21 @@ pub mod tests {
         let client_1 = Client::new(server_pk, addr, incoming_tx_1);
         // connection attempts should be set to 0 after successful connection
         set_connection_attempts(&client_1, 3);
-        let client_future_1 = client_1.clone().spawn(client_sk_1, client_pk_1);
+        let client_future_1 = client_1.clone().spawn(client_sk_1, client_pk_1)
+            .map_err(|e| Error::new(ErrorKind::Other, e.compat()));
```

Comment:
Why?

Reply:
It seems because it gets joined with other futures with diffirent errors. But I'd suggest to use `failure::Error` in such cases - it can be easily constructed from almost any error with `.map_err(Error::from)`.